fenixedu-ist-integration/src/main/java/pt/ist/fenixedu/integration/api/EmployeeAssiduityResource.java

```diff
@@ -64,7 +66,17 @@ private String respond(final Function<User, JsonObject> function) {
     }
 
     private LocalDate parse(final String date) {
-        return DateTimeFormat.forPattern("yyyy-MM-dd").parseLocalDate(date);
+        try {
+            return DateTimeFormat.forPattern(dayPattern).parseLocalDate(date);
+        } catch (Exception e) {
+            throw newApplicationError(Response.Status.PRECONDITION_FAILED, "format_error", "day must be " + dayPattern);
+        }
     }
 
+    private WebApplicationException newApplicationError(Response.Status status, String error, String description) {
+        JsonObject errorObject = new JsonObject();
+        errorObject.addProperty("error", error);
+        errorObject.addProperty("description", description);
+        return new WebApplicationException(Response.status(status).entity(errorObject.toString()).build());
+    }
```

Comment:
I don't think we have any exceptions with registered ExceptionMappers that can be used in here. If there were any, they would probably be used in the FenixAPIv1. There are a couple regarding JSON mapping & parsing, but none seems to use this format (and this isn't JSON parsing).